provider/pkg/provider/deployment_settings.go

```
@@ -0,0 +1,585 @@
+	if err != nil {
+		return nil, err
+	}
+
+	dsInput := PulumiServiceDeploymentSettingsInput{
+		Stack:              stack,
+		DeploymentSettings: *settings,
+	}
+
+	properties, err := plugin.MarshalProperties(
+		dsInput.ToPropertyMap(),
+		plugin.MarshalOptions{},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &pulumirpc.ReadResponse{
+		Id:         req.Id,
+		Properties: properties,
+		Inputs:     properties,
+	}, nil
+}
+
+func (ds *PulumiServiceDeploymentSettingsResource) Delete(req *pulumirpc.DeleteRequest) (*pbempty.Empty, error) {
+	ctx := context.Background()
+	inputsMap, err := plugin.UnmarshalProperties(req.GetProperties(), plugin.MarshalOptions{KeepUnknowns: true, SkipNulls: true})
+	if err != nil {
+		return nil, err
+	}
+	inputs := ds.ToPulumiServiceDeploymentSettingsInput(inputsMap)
+	if err != nil {
+		return nil, err
+	}
+	err = ds.client.DeleteDeploymentSettings(ctx, inputs.Stack)
+	if err != nil {
+		return nil, err
+	}
+	return &pbempty.Empty{}, nil
+}
+
+func (ds *PulumiServiceDeploymentSettingsResource) Create(req *pulumirpc.CreateRequest) (*pulumirpc.CreateResponse, error) {
+	ctx := context.Background()
+	inputsMap, err := plugin.UnmarshalProperties(req.GetProperties(),
+		plugin.MarshalOptions{KeepUnknowns: true, SkipNulls: true, KeepSecrets: true})
+	if err != nil {
+		return nil, err
+	}
+	inputs := ds.ToPulumiServiceDeploymentSettingsInput(inputsMap)
+	settings := inputs.DeploymentSettings
+	err = ds.client.CreateDeploymentSettings(ctx, inputs.Stack, settings)
+	if err != nil {
+		return nil, err
+	}
+	return &pulumirpc.CreateResponse{
+		Id:         path.Join(inputs.Stack.OrgName, inputs.Stack.ProjectName, inputs.Stack.StackName),
+		Properties: req.GetProperties(),
+	}, nil
+}
+
+func (ds *PulumiServiceDeploymentSettingsResource) Update(req *pulumirpc.UpdateRequest) (*pulumirpc.UpdateResponse, error) {
+	// For simplicity, all updates are destructive, so we just call Create.
+	return &pulumirpc.UpdateResponse{}, nil
```

This should return an error.